airflow/providers/amazon/aws/hooks/redshift_cluster.py

```diff
@@ -285,6 +285,47 @@ async def resume_cluster(
             except botocore.exceptions.ClientError as error:
                 return {"status": "error", "message": str(error)}
 
+    async def delete_cluster(
+        self,
+        cluster_identifier: str,
+        skip_final_cluster_snapshot: bool = True,
+        final_cluster_snapshot_identifier: str | None = None,
+        poll_interval: float = 5.0,
+    ) -> dict[str, Any]:
+        """
+        Connects to the AWS redshift cluster via aiobotocore and
+        deletes the cluster based on the cluster_identifier passed
+
+        :param cluster_identifier: unique identifier of a cluster
+        :param skip_final_cluster_snapshot: determines cluster snapshot creation
+        :param final_cluster_snapshot_identifier: name of final cluster snapshot
+        :param poll_interval: polling period in seconds to check for the status
+        """
+        try:
+            final_cluster_snapshot_identifier = final_cluster_snapshot_identifier or ""
+
+            async with await self.get_client_async() as client:
+                response = await client.delete_cluster(
+                    ClusterIdentifier=cluster_identifier,
+                    SkipFinalClusterSnapshot=skip_final_cluster_snapshot,
+                    FinalClusterSnapshotIdentifier=final_cluster_snapshot_identifier,
+                )
+                status = response["Cluster"]["ClusterStatus"] if response and response["Cluster"] else None
+                if status == "deleting":
```

Comment:
If the initial call to `delete_cluster` hits the `InvalidClusterStateFault` Exception, then this function will not attempt to retry. The `InvalidClusterStateFault` is only seen in the first few attempts, when the cluster is not ready to accept new instructions even if it is reporting its state as `available`.

Reply:
this changes has been removed form this PR